### PR TITLE
Basic memoryTaint Outline 

### PR DESCRIPTION
**Pull request by sandhrabino**

### CODE REVIEW

**Comment by R3x** [comment]
You have committed an extra file!

File: TaintTracker/TaintTracker.cpp [hunk 1]
```diff
@@ -4,19 +4,36 @@
...
-bool isLibraryFunction() {
-    // TODO : Figure out to do this
-    return true;
+bool isLibraryFunction( UINT64 startAddr) {
+    if (startAddr >= LIBC_BASE ){
+      return true;
+    }
+    else {
+      return false;
+    }
```
Comment (R3x) [inline TaintTracker/TaintTracker.cpp]: I think @Freakston or @adityavardhanpadala has a better solution. Please discuss in the issue

File: TaintTracker/TaintTracker.cpp [hunk 2]
```diff
@@ -28,16 +45,31 @@ bool isLibraryFunction() {
...
+    bool check_taint = false;
+    list<struct range>::iterator i;
+    struct range taint;
+
+    for(i = bytesTainted.begin(); i != bytesTainted.end(); i++) {
+				if (((struct range) *i).start <= addr && ((struct range) *i).size + ((struct range) *i).start >= addr) {
+						check_taint = true;
+						break;
+				}
+		}
```
Comment (R3x) [inline TaintTracker/TaintTracker.cpp]: Indentation issues here

**Review by R3x — CHANGES_REQUESTED** [review]
Let's take a step back before committing the change.

I created a different template for the taint tracking. So the one that you pushed needs to be changed.

File: TaintTracker/TaintTracker.cpp (hunk 1 above)
Comment (sandhrabino) [reply TaintTracker/TaintTracker.cpp]: Okay I will discuss with them

**Pull request merged**